### PR TITLE
Lesson 7 dobricsongor

### PR DESCRIPTION
**Pull request by dobricsongor**

### CODE REVIEW

File: lesson_6/assignment.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+### Let's practice
```
Comment (mariuspodean) [inline lesson_6/assignment.md]: This is the requirements file from the previous assignment. Please replace it with the current one :)

File: lesson_7/dobricsongor/decorator.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Create a decorator called uppercase that will uppercase the result
+
+def uppercase(fnc):
+    
+    def inner_func(given_text):
+        return(fnc(given_text).upper())
```
Comment (mariuspodean) [inline lesson_7/dobricsongor/decorator.py]: The following produce same results:

```python3
return(fnc(given_text).upper())

return fnc(given_text).upper()
```

Comment (mariuspodean) [reply lesson_7/dobricsongor/decorator.py]: You could drop the extra parentheses

File: lesson_7/dobricsongor/decorator.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        else:
+            print(f'The operation divide of {first_num} and {second_num} cannot be done!') 
+
+    return inner_func
+
+@safe_divide
+def divide(first_number, second_number):
+    return first_number / second_number
+
+divide(10.3, 2)
```
Comment (mariuspodean) [inline lesson_7/dobricsongor/decorator.py:39]: What would happen if we will try the following division: 3/0 ?

File: lesson_7/dobricsongor/decorator.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+#------------------------------------------------------------------------------
+
+#Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result.
+
+def safe_divide(fnc):
+
+    def inner_func(first_num,second_num):
+        if isinstance(first_num, (int,float)) and isinstance(second_num, (int,float)):
```
Comment (mariuspodean) [inline lesson_7/dobricsongor/decorator.py]: Most of the time we don't check types in Python, we simply check that certain behaviour is present on a Python object

**Review by mariuspodean — COMMENTED** [review]
Good job! I've added some observations

File: lesson_7/dobricsongor/decorator.py (hunk 3 above)
Comment (dobricsongor) [reply lesson_7/dobricsongor/decorator.py:39]: Ma gandeam ca division 3/0 se poate face, doar ca rezultatul e 0. 
Rezolv :)

File: lesson_7/dobricsongor/decorator.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+#------------------------------------------------------------------------------
+
+#Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result.
+
+def safe_divide(fnc):
+
+    def inner_func(first_num,second_num):
+        #if isinstance(first_num, (int,float)) and isinstance(second_num, (int,float)): 
+            if second_num != 0 and first_num != 0:
```
Comment (Liitle) [inline lesson_7/dobricsongor/decorator.py]: Implementarea decoratorului este buna. O simgura observatie, putem sa verificam doar daca `second_num !=0` pentru ca `3/0` este o operatie invalida, dar `0/3` este o operatie valida care are rezultatul `0` 

```
In [1]: 0/3                                                                                                                           
Out[1]: 0.0

In [2]: 3/0                                                                                                                           
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-2-f6cc6d14333b> in <module>
----> 1 3/0

ZeroDivisionError: division by zero

```

**Review by Liitle — CHANGES_REQUESTED** [review]
super tema 🥳 , good job! am lasat o mica observatie mai jos

**Pull request merged**